t/35-script_clone_job.t

```diff
@@ -129,4 +129,60 @@ subtest 'asset download' => sub {
     ok(-f "$temp_assetdir/iso/bar.iso", 'foo touched');
 };
 
+subtest 'get 2 nodes HA cluster with get_deps' => sub {
+    my %node1_settings = (
+        NAME          => '00007936-sle-15-SP3-Online-x86_64-Build67.1-sles4sap_hana_node01@64bit-sap',
+        TEST          => 'sles4sap_hana_node01',
+        HDD_1         => 'SLE-15-SP3-x86_64-Build67.1-sles4sap-gnome.qcow2',
+        HDDSIZEGB     => 60,
+        PARALLEL_WITH => 'sles4sap_hana_supportserver@64bit-2gbram',
+        NICTYPE       => 'tap',
+        WORKER_CLASS  => 'tap,qemu_x86_64',
+    );
+    my %node2_settings = (
+        NAME          => '00007935-sle-15-SP3-Online-x86_64-Build67.1-sles4sap_hana_node02@64bit-sap',
+        TEST          => 'sles4sap_hana_node02',
+        HDD_1         => 'SLE-15-SP3-x86_64-Build67.1-sles4sap-gnome.qcow2',
+        HDDSIZEGB     => 60,
+        PARALLEL_WITH => 'sles4sap_hana_supportserver@64bit-2gbram',
+        NICTYPE       => 'tap',
+        WORKER_CLASS  => 'tap,qemu_x86_64',
+    );
+    my %supportserver_settings = (
+        NAME         => "00007934-sle-15-SP3-Online-x86_64-Build67.1-sles4sap_hana_supportserver@64bit-2gbram",
+        TEST         => 'sles4sap_hana_supportserver',
+        HDD_1        => 'openqa_support_server_sles12sp3.x86_64.qcow2',
+        HDDSIZEGB    => 60,
+        NICTYPE      => 'tap',
+        WORKER_CLASS => 'tap,qemu_x86_64',
+    );
+    my %node1_job = (
+        settings => \%node1_settings,
+        id => 7936,
+        parents => (
+            Parallel => (7934),
+        ),
```

Comment:
```suggestion
        parents => {
            Parallel => [7934],
        },
```

The same counts for the other occurrences. Note that `()` is used in Perl to create a "list" (not sure about the terminology). From a "list" other stuff like hashes and arrays are initialized. However, if you create nested structures, "lists" are just "unpacked"/"expanded" (still not sure about the terminology). So `(…, parents => (Parallel => (7834)))` is actually equivalent to just `(…, parents => Parallel => 7834)` which is not what you want. So you need to use `{}` and `[]` to create a hash/array-ref explicitly.

Reply:
Changed, thx!